File: Arithmetic_FastHCF.cpp
// Arithmetic.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include "FastHCF.h"

#include <iostream>
#include <sstream>
#include <vector>

using namespace std;

int main() {
    try {
        cout << "\n    Welcome to our demonstration sequels\n";
        cout << "Hope you enjoy (and follow) the lessons.\n\n";

        vector<unsigned> set;

        /*
         * Find HCF.
         */
        string render_result = "The HCF of ";
        stringstream aux;

        vector<unsigned> set;
        set = {30, 48, 54};

        for (int number : set) {
            aux.str("");
            aux << number;
            render_result += aux.str() + "; ";
        }
        render_result += "is: ";

        // Use fast HCF
        FastHCF h_c_f(set);
        cout << "\n\n" << render_result << h_c_f.getHCF() << "\n";

    } catch (exception& e) {
        cout << "\n" << e.what() << "\n";
    }
    return 0;
}

File: Arithmetic_FastHCF.java
package arithmetic;

import java.util.ArrayList;
import java.util.List;

public class Arithmetic {

    public static void main(String[] args) {
        System.out.println("Welcome to our demonstration sequels");
        System.out.println("Hope you enjoy (and follow) the lessons.");
        System.out.println("");

        List<Integer> set;
        set = new ArrayList<>();
        set.add(30);
        set.add(48);
        set.add(54);

        String render_result = "The HCF of ";
        for (int number : set) {
            render_result += number + "; ";
        }
        render_result += "is:  ";

        //Use fast copy
        FastHCF h_c_f;
        h_c_f = new FastHCF(set);
        System.out.println(render_result + h_c_f.getHCF());
    }

}
